core-tests/src/test/java/org/mule/runtime/core/internal/processor/strategy/AbstractProcessingStrategyTestCase.java

```diff
@@ -502,33 +469,37 @@ protected void testAsyncCpuLightNotificationThreads(AtomicReference<Thread> befo
     flow = flowBuilder.get().processors(annotatedAsyncProcessor).build();
     flow.initialise();
     flow.start();
-    process(flow, testEvent());
+    processFlow(testEvent());
   }
 
-  protected void testBackPressure(BackPressureStrategy backPressureStrategy, Predicate<Integer> processedAssertion,
-                                  Predicate<Integer> rejectedAssertion, Predicate<Integer> totalAssertion)
+  protected void testBackPressure(BackPressureStrategy backPressureStrategy, Matcher<Integer> processedAssertion,
+                                  Matcher<Integer> rejectedAssertion, Matcher<Integer> totalAssertion)
       throws MuleException {
     if (mode.equals(SOURCE)) {
 
-      TriggerableMessageSource triggerableMessageSource = new TriggerableMessageSource(backPressureStrategy);
+      triggerableMessageSource = new TriggerableMessageSource(backPressureStrategy);
       flow =
-          flowBuilder.get().source(triggerableMessageSource).processors(asList(cpuLightProcessor, new ThreadTrackingProcessor() {
-
-            @Override
-            public InternalEvent process(InternalEvent event) throws MuleException {
-              try {
-                Thread.sleep(3);
-              } catch (InterruptedException e) {
-                throw new RuntimeException(e);
-              }
-              return super.process(event);
-            }
-
-            @Override
-            public ProcessingType getProcessingType() {
-              return BLOCKING;
-            }
-          })).maxConcurrency(2).build();
+          flowBuilder.get()
+              .source(triggerableMessageSource)
+              .processors(asList(cpuLightProcessor, new ThreadTrackingProcessor() {
+
+                @Override
+                public InternalEvent process(InternalEvent event) throws MuleException {
+                  try {
+                    Thread.sleep(3);
```

Comment:
the 3 is still here

Reply:
I've ensured this is as reliable as possible to avoid flaky tests.  I attempted to do this with a latch, but not only is it very hard, it's not a realistic back-pressure test that simulates a slow-consumer rather than latched processors.  If we see this is flaky then can revisit but i'm convinced that given loose assertions it won't be an issue.